independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralGenerator.java

```diff
@@ -142,45 +137,12 @@ static void createAnnotationLiteral(ClassOutput classOutput, ClassInfo annotatio
             if (value == null) {
                 value = method.defaultValue();
             }
-            ResultHandle retValue = null;
             if (value == null) {
-                switch (method.returnType().kind()) {
-                    case CLASS:
-                    case ARRAY:
-                        retValue = valueMethod.loadNull();
-                        break;
-                    case PRIMITIVE:
-                        PrimitiveType primitiveType = method.returnType().asPrimitiveType();
-                        switch (primitiveType.primitive()) {
-                            case BOOLEAN:
-                                retValue = valueMethod.load(false);
-                                break;
-                            case BYTE:
-                            case SHORT:
-                            case INT:
-                                retValue = valueMethod.load(0);
-                                break;
-                            case LONG:
-                                retValue = valueMethod.load(0L);
-                                break;
-                            case FLOAT:
-                                retValue = valueMethod.load(0.0f);
-                                break;
-                            case DOUBLE:
-                                retValue = valueMethod.load(0.0d);
-                                break;
-                            case CHAR:
-                                retValue = valueMethod.load('\u0000');
-                                break;
-                        }
-                        break;
-                    default:
-                        break;
-                }
-            } else {
-                retValue = loadValue(valueMethod, value, annotationClass, method);
+                throw new NullPointerException(String.format(
+                        "Value is not set for %s.%s(). Most probably an older version of Jandex was used to index an application dependency. Make sure that Jandex 2.1+ is used.",
+                        method.declaringClass().name(), method.name()));
```

Comment:
So we don't need to handle the primitive case anymore?

Reply:
We do handle primitives in `AnnotationLiteralGenerator.loadValue()`. I've just removed the part which sets the values to the default JLS values - this wouldn't help much and the produced literal would be incorrect.